gdal/frmts/netcdf/netcdflayer.cpp

```diff
@@ -2455,42 +2721,81 @@ OGRErr netCDFLayer::CreateField(OGRFieldDefn *poFieldDefn, int /* bApproxOK */)
     case OFTDate:
     {
         nType = NC_INT;
-        status = nc_def_var(m_nLayerCDFId, pszVarName, nType, 1, &nMainDimId,
-                            &nVarID);
+
+        if(!m_bLegacyCreateMode)
```

Comment:
see above remarks. could be merged in a single code path with a passthrough vcdf implementation